### PR TITLE
fix: Use Database Timeout setting defined in the configuration

### LINKED ISSUE #4486: Use Database Timeout setting defined in the configuration
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

 The issue is located in all services using database

### Is this a regression?

No

### Description and Minimal Reproduction [**REQUIRED**]
We define Database Timeout in the configuration, but not using it.
edgex-go services should use timeout value as we do in app service.

### PR DESCRIPTION
**Pull request by jinlinGuan**
close #4486

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Review by cloudxxx8 — DISMISSED** [review]
LGTM

File: internal/pkg/db/redis/client.go [hunk 1]
```diff
@@ -42,10 +42,16 @@ type CoreDataClient struct {
 
 // Return a pointer to the Redis client
 func NewClient(config db.Configuration, lc logger.LoggingClient) (*Client, error) {
+	var retErr error
 	once.Do(func() {
 		connectionString := fmt.Sprintf("%s:%d", config.Host, config.Port)
+		connectTimeout, err := time.ParseDuration(config.Timeout)
+		if err != nil {
+			retErr = fmt.Errorf("config.Timeout failed to parse: %v", err)
```
Comment (lenny-intel) [inline internal/pkg/db/redis/client.go:50]:
```suggestion
			retErr = fmt.Errorf("configured database timeout failed to parse: %v", err)
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/pkg/db/redis/client.go (hunk 1 above)
Comment (jinlinGuan) [reply internal/pkg/db/redis/client.go]: updated

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**